### DISCUSSION TITLE
Is there a DSL way to apply a Miller verb in loop?

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#457] [Q&A] [answered]
Hi all,
I would like to apply fill-down verb to all fields of a CSV file.
To do it I could build a bash for loop.

Is there a DSL way to run something like the code below?

```
  for (key in $*) {
   fill-down key
  }
```

Thank you

**Comment by johnkerl** [comment, marked as answer]
@aborruso DSL expressions don't call verbs per se -- however, under the hood a verb is a command-line wrapper around some code in a subroutine. That same code can have a second, DSL wrapper around it and, voilà, a function which is callable from the DSL.

Already-existing examples include `seg2gmt` which is a verb and a function; likewise the verb `clean-whitespace` and the function `clean_whitespace`.

So, we could do something similar here.

However, the `fill-down` verb is a bit trickier since it retains state from the previous record -- making it not a function solely of the current record.

For your use-case, which is `fill-down` over all fields -- maybe we could instead simply augment the existing `fill-down` verb to accept a `--all` flag -- ?

**Reply by aborruso** [reply]
It would be great.

I have solved using bash:

mlr --csv put -q 'NR == 1 {for (key in $*) {if (key=~"^.{4}$") {print key}}}' ./input.csv |
    while read line; do
      mlr -I --csv fill-down -f "$line" ./areeStorico_wide_fill_down.csv
    done
